### PR TITLE
basic routes and plp

### PR DESCRIPTION
**Pull request by ravikumar0403**
Change log:
- added basic routes
- added product context
- added product listing page

### CODE REVIEW

**Review by yashpatel1797 — APPROVED** [review]
products are not loading. tried 2&3 times.

**Comment by ravikumar0403** [comment]
> products are not loading. tried 2&3 times.

That was a issue on the server which is fixed now. Thanks for mentioning it.

File: src/components/Filters.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      <div className="my-1">
+        <p className="fs-2">Sort by</p>
+        <select className="input">
+          <option value="" defaultValue>
+            sort by
+          </option>
+          <option value="price_asc">Price: low to high</option>
+          <option value="price_desc">Price: high to low</option>
+          <option value="rating_asc">Rating: low to high</option>
+          <option value="rating_desc">Rating: high to low</option>
```
Comment (shashikantmaurya0007) [inline src/components/Filters.js:17]: filter on rating should be added under separate select input

File: src/components/Navbar.js [hunk 2]
```diff
@@ -7,12 +8,18 @@ export const Navbar = () => {
...
+            <img loading="lazy" src={logo} alt="logo" />
+          </div>
+        </Link>
         <ul className="nav-menu">
-          <li className="nav-menu-item mx-1">Home</li>
-          <li className="nav-menu-item mx-1">Shop Now</li>
+          <Link to="/">
+            <li className="nav-menu-item mx-1">Home</li>
+          </Link>
+          <Link to="products">
```
Comment (shashikantmaurya0007) [inline src/components/Navbar.js:20]: i guess you missed '/' here

File: src/components/Navbar.js [hunk 3]
```diff
@@ -23,18 +30,23 @@ export const Navbar = () => {
...
-            </div>
-          </li>
-          <li className="nav-menu-item mx-1">
-            <div className="badge-container">
-              <i className="fs-2 fa fa-cart-shopping"></i>
-              <div className="badge top right bg-primary">5</div>
-            </div>
-          </li>
+          <Link to="wishlist">
+            {" "}
```
Comment (shashikantmaurya0007) [inline src/components/Navbar.js:34]: remove this line

**Review by shashikantmaurya0007 — APPROVED** [review]
I have suggested few changes .your code is really well structured. I liked how you have fetch data in customProvider. :)

File: src/components/Filters.js (hunk 1 above)
Comment (ravikumar0403) [reply src/components/Filters.js:17]: filters functionality is added separately. This is for sorting based on rating

File: src/components/Navbar.js (hunk 2 above)
Comment (ravikumar0403) [reply src/components/Navbar.js:20]: thanks, I'll change this

**Pull request merged**